classes/announcement/Announcement.php

```diff
@@ -135,6 +135,16 @@ public function save(array $options = [])
             $this->handleImageUpload();
         }
 
+        // If there is no new image and image data exists in DB and it's now removed
+        // need to delete the image for this announcement model instance
+        if (!$hasNewImage && !$this?->image && $this->fresh()->image) {
+            $this->deleteImage();
+            DB::table($this->getSettingsTable())
```

Comment:
I would expect that settings are removed together with the main model, isn't this happening?

Reply:
This handle a case when at updating an announcement image is deleted and that is the only update .

Reply:
If it's the only case, I'm ok with it)